tests/test_seasonal_date_factor.R

```diff
@@ -0,0 +1,120 @@
+library(climdex.pcic)
+library(RUnit)
+# Test that NA masks work with seasonal aggregation and that indices are correctly calculated.
+
+# Setup
+wd <- "./"
+ClimVars.tmax <- file.path(wd, "1018935_MAX_TEMP.csv")
+ClimVars.tmin <- file.path(wd, "1018935_MIN_TEMP.csv")
+ClimVars.prec <- file.path(wd, "1018935_ONE_DAY_PRECIPITATION.csv")
+
+datacol <- list(tmax = "MAX_TEMP", tmin = "MIN_TEMP", prec = "ONE_DAY_PRECIPITATION")
+ci.csv <- climdexInput.csv(ClimVars.tmax, ClimVars.tmin, ClimVars.prec, data.columns = datacol, base.range = c(1981, 1990))
+
+unique_seasons <- unique(ci.csv@date.factors$seasonal)
+unique_months <- unique(ci.csv@date.factors$monthly)
+
+climdex.pcic.test.seasonal.na.cases <- function() {
+  var.list <- c("tmax", "tmin", "prec", "tavg")
+  for (var in var.list){
+    # Test 1.1: Check if the first season is NA based on the start of the time series
+    # - If the first month of the date series is not a multiple of 3, expect the first season to be NA
+    checkTrue((as.integer(format(ci.csv@dates[1], "%m")) %% 3 != 0) && is.na(ci.csv@namasks$seasonal[[var]][1]))
```

Comment:
There's a lot going on in these checks. Any way that you can break this up a bit more for readability?